### PR TITLE
Timers create validation

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: pkg/systemd/services/timer-dialog.jsx [hunk 1]
```diff
@@ -74,18 +74,76 @@ const CreateTimerDialogBody = ({ owner }) => {
...
-        validationFailed.name = true;
-    if (!description.trim().length)
-        validationFailed.description = true;
-    if (!command.trim().length || commandNotFound)
-        validationFailed.command = true;
-    if (!/^[0-9]+$/.test(delayNumber))
-        validationFailed.delayNumber = true;
+    const validate = async (event) => {
+        if (!event)
+            return;
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:83]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 2]
```diff
@@ -74,18 +74,76 @@
...
+        case 'servicename': {
+            if (value.trim().length && /^[a-zA-Z0-9:_.@-]+$/.test(value))
+                notValid = false;
+            break;
+        }
+        case 'description': {
+            if (value.trim().length)
+                notValid = false;
+            break;
+        }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:99]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 3]
```diff
@@ -74,18 +74,76 @@
...
+        case 'command': {
+            if (!command.trim().length)
+                break;
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:102]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 4]
```diff
@@ -74,18 +74,76 @@
...
+            const command_parts = command.split(" ");
+            setInProgress(true);
+            try {
+                await cockpit.spawn(["test", "-f", command_parts[0]], { err: "ignore" });
+                notValid = false;
+            } catch {
+                setCommandNotFound(true);
+            }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:110]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 5]
```diff
@@ -74,18 +74,76 @@
...
+            if (!value.trim().length)
+                break;
+            setInProgress(true);
+            try {
+                await cockpit.spawn(["systemd-analyze", "calendar", value.trim()]);
+                notValid = false;
+            } catch {}
+            setInProgress(false);
+            break;
+        }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:129]: These 16 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 6]
```diff
@@ -74,18 +74,76 @@
...
+        // invalidate base form items.
+        default: {
+            setValidationFailed(validationFailed => ({
+                servicename: true,
+                description: true,
+                command: true
+            }));
+            return;
+        }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:138]: These 8 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 7]
```diff
@@ -106,45 +164,98 @@
...
+    const onSubmit = (event) => {
+        if (!event)
+            return false;
+
+        let failed = false;
+        let failMessage = _("Please complete the form");
 
-        if (event)
-            event.preventDefault();
+        event.preventDefault();
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:174]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 8]
```diff
@@ -106,45 +164,98 @@
...
-                            .then(Dialogs.close, exc => {
-                                setDialogError(exc.message);
-                                setInProgress(false);
-                            });
-                })
-                .catch(() => {
-                    setCommandNotFound(true);
+        create_timer({ name, description, command, delay, delayUnit, delayNumber, repeat, specificTime, repeatPatterns, owner })
+                .then(Dialogs.close, exc => {
+                    setDialogError(exc.message);
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:235]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 9]
```diff
@@ -268,49 +398,72 @@
...
                                 <FormGroup label={label} key={item.key}
-                                           validated={validated}
-                                           helperTextInvalid={helperTextInvalid}>
+                                           validated={repeatPatterns[idx].validated}
+                                           helperTextInvalid={repeatPatterns[idx].helperTextInvalid}>
                                     <Flex className="specific-repeat-group" data-index={idx}>
                                         {repeat == "minutely" &&
                                             <TextInput className='delay-number'
                                                        id={repeat}
+                                                       isRequired
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:437]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 10]
```diff
@@ -268,49 +398,72 @@
...
+                                                       pattern='^[0-9]+$'
+                                                       type='number'
+                                                       min='0'
+                                                       max='59'
+                                                       validated={repeatPatterns[idx].validated} />
                                         }
                                         {repeat == "hourly" && <>
                                             <TextInput className='delay-number'
+                                                       id={repeat}
+                                                       isRequired
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:454]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 11]
```diff
@@ -268,49 +398,72 @@
...
                                                        value={repeatPatterns[idx].minute}
+                                                       onBlur={repeatValidate}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:456]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 12]
```diff
@@ -268,49 +398,72 @@
...
                                                            const arr = [...repeatPatterns];
                                                            arr[idx].minute = minute;
                                                            setRepeatPatterns(arr);
                                                        }}
-                                                       validated={submitted && validationFailedMinute ? "error" : "default"} />
+                                                       pattern='^[0-9]+$'
+                                                       type='number'
+                                                       min='0'
+                                                       max='59'
+                                                       validated={repeatPatterns[idx].validated} />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:466]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 13]
```diff
@@ -353,15 +506,26 @@
...
+                                                        arr[idx].date = str;
+                                                        setRepeatPatterns(arr);
+                                                    }
+                                                }}
+                                                appendTo={() => document.body}
+                                                buttonAriaLabel={_("Toggle date picker")}
+                                                locale={timeformat.dateFormatLang()}
+                                                weekStart={timeformat.firstDayOfWeek()}
+                                                // isRequired // No Required flag for DatePicker
+                                            />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:528]: These 20 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 14]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
     );
 
-    function onSubmit(event) {
-        setSubmitted(true);
+    const onSubmit = (event) => {
+        if (!event)
+            return false;
+
+        let failed = false;
+        let failMessage = _("Please complete the form");
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:172]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 15]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
 
-        if (event)
-            event.preventDefault();
+        event.preventDefault();
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:174]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 16]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                // Need yearly check as it doesnt have a default
+                switch (repeat) {
+                case 'yearly': {
+                    const check = repeatPatterns.find(pattern => {
+                        return !pattern.date;
+                    });
+                    if (check)
+                        failed = true;
+                    break;
+                }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:215]: These 40 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 17]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                }
+                break;
+            }
+            // Nothing to validate at this time.
+            case 'system-boot': {
+                break;
+            }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:222]: These 6 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 18]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+            }
+        }
+        if (failed) {
+            setDialogError(failMessage);
+            setValidForm(false);
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:227]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 19]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
             return false;
+        }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:229]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 20]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
-                            .then(Dialogs.close, exc => {
-                                setDialogError(exc.message);
-                                setInProgress(false);
-                            });
-                })
-                .catch(() => {
-                    setCommandNotFound(true);
+        create_timer({ name, description, command, delay, delayUnit, delayNumber, repeat, specificTime, repeatPatterns, owner })
+                .then(Dialogs.close, exc => {
+                    setDialogError(exc.message);
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:235]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 21]
```diff
@@ -106,45 +164,98 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                     setInProgress(false);
                 });
 
         return false;
-    }
+    };
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:240]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 22]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
-                               validated={submitted && validationFailed.name ? "error" : "default"}
+                               validated={validationFailed.servicename ? "error" : "default"}
+                               onBlur={validate}
                                onChange={setName} />
                 </FormGroup>
                 <FormGroup label={_("Description")}
+                           isRequired
                            fieldId="description"
-                           validated={submitted && validationFailed.description ? "error" : "default"}
+                           validated={validationFailed.description ? "error" : "default"}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:284]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 23]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                            helperTextInvalid={_("This field cannot be empty")}>
                     <TextInput id='description'
+                               isRequired
                                value={description}
-                               validated={submitted && validationFailed.description ? "error" : "default"}
+                               validated={validationFailed.description ? "error" : "default"}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:289]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 24]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                               onBlur={validate}
                                onChange={setDescription} />
                 </FormGroup>
                 <FormGroup label={_("Command")}
+                           isRequired
                            fieldId="command"
-                           validated={submitted && validationFailed.command ? "error" : "default"}
+                           validated={validationFailed.command ? "error" : "default"}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:296]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 25]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                            helperTextInvalid={commandNotFound ? _("Command not found") : _("This field cannot be empty")}>
                     <TextInput id='command'
+                               isRequired
                                value={command}
-                               validated={submitted && validationFailed.command ? "error" : "default"}
+                               validated={validationFailed.command ? "error" : "default"}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:301]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 26]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                 <FormGroup label={_("Trigger")} hasNoPaddingTop>
                     <Flex>
                         <Radio value="specific-time"
                                id="specific-time"
                                name="boot-or-specific-time"
-                               onChange={() => setDelay("specific-time")}
+                               onChange={() => {
+                                   setDelay("specific-time");
+                                   formCheck();
+                               }}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:313]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 27]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                isChecked={delay == "specific-time"}
                                label={_("At specific time")} />
                         <Radio value="system-boot"
                                id="system-boot"
                                name="boot-or-specific-time"
-                               onChange={() => setDelay("system-boot")}
+                               onChange={() => {
+                                   setDelay("system-boot");
+                                   formCheck();
+                               }}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:322]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 28]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                isChecked={delay == "system-boot"}
                                label={_("After system boot")} />
                     </Flex>
                     { delay == "system-boot" &&
                     <FormGroup className="delay-group"
+                               isRequired
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:328]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 29]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                label={_("Delay")}
-                               validated={submitted && validationFailed.delayNumber ? "error" : "default"}
+                               validated={validationFailed['delay-number'] ? "error" : "default"}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:330]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 30]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                helperTextInvalid={_("Delay must be a number")}>
                         <Flex>
                             <TextInput className="delay-number"
+                                       aria-label={_("Delay in seconds")}
+                                       isRequired
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:335]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 31]
```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                        value={delayNumber}
-                                       validated={submitted && validationFailed.delayNumber ? "error" : "default"}
+                                       validated={validationFailed['delay-number'] ? "error" : "default"}
+                                       onBlur={validate}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:338]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 32]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                                    break;
+                                }
+                                case 'minutely': {
+                                    helperTextInvalid = _("Second needs to be a number between 0-59");
+                                    break;
+                                }
+                                case 'yearly': {
+                                    helperTextInvalid = _("Please enter valid date.  Example YYYY-MM-DD");
+                                    break;
+                                }
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:416]: These 16 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 33]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                                const validated = !event.target.checkValidity() ? "error" : "default";
+                                const arr = [...repeatPatterns];
+                                arr[idx].helperTextInvalid = helperTextInvalid;
+                                arr[idx].validated = validated;
+                                setRepeatPatterns(arr);
+                                if (validated == 'default') {
+                                    setDialogError();
+                                    formCheck();
+                                }
+                            };
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:427]: These 10 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 34]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
 
                             return (
                                 <FormGroup label={label} key={item.key}
-                                           validated={validated}
-                                           helperTextInvalid={helperTextInvalid}>
+                                           validated={repeatPatterns[idx].validated}
+                                           helperTextInvalid={repeatPatterns[idx].helperTextInvalid}>
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:432]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 35]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                     <Flex className="specific-repeat-group" data-index={idx}>
                                         {repeat == "minutely" &&
                                             <TextInput className='delay-number'
                                                        id={repeat}
+                                                       isRequired
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:437]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 36]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                                        value={repeatPatterns[idx].second}
+                                                       onBlur={repeatValidate}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:439]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 37]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                                            const arr = [...repeatPatterns];
                                                            arr[idx].second = second;
                                                            setRepeatPatterns(arr);
                                                        }}
-                                                       validated={submitted && validationFailedSecond ? "error" : "default"} />
+                                                       pattern='^[0-9]+$'
+                                                       type='number'
+                                                       min='0'
+                                                       max='59'
+                                                       validated={repeatPatterns[idx].validated} />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:449]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 38]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                         }
                                         {repeat == "hourly" && <>
                                             <TextInput className='delay-number'
+                                                       id={repeat}
+                                                       isRequired
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:454]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 39]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                                        value={repeatPatterns[idx].minute}
+                                                       onBlur={repeatValidate}
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:456]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 40]
```diff
@@ -268,49 +398,72 @@ const CreateTimerDialogBody = ({ owner }) => {
...
                                                            const arr = [...repeatPatterns];
                                                            arr[idx].minute = minute;
                                                            setRepeatPatterns(arr);
                                                        }}
-                                                       validated={submitted && validationFailedMinute ? "error" : "default"} />
+                                                       pattern='^[0-9]+$'
+                                                       type='number'
+                                                       min='0'
+                                                       max='59'
+                                                       validated={repeatPatterns[idx].validated} />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:466]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/services/timer-dialog.jsx [hunk 41]
```diff
@@ -353,15 +506,26 @@ const CreateTimerDialogBody = ({ owner }) => {
...
+                                                        arr[idx].date = str;
+                                                        setRepeatPatterns(arr);
+                                                    }
+                                                }}
+                                                appendTo={() => document.body}
+                                                buttonAriaLabel={_("Toggle date picker")}
+                                                locale={timeformat.dateFormatLang()}
+                                                weekStart={timeformat.firstDayOfWeek()}
+                                                // isRequired // No Required flag for DatePicker
+                                            />
```
Comment (cockpituous) [inline pkg/systemd/services/timer-dialog.jsx:528]: These 20 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
@martinpitt you should not get this.

**Comment by martinpitt** [comment]
@jelly This is stale/conflicty, and still needs a lot of work to cover it in tests. Are you still interested in working on this, or should we close this ("put it into the attic")?

**Comment by jelly** [comment]
> @jelly This is stale/conflicty, and still needs a lot of work to cover it in tests. Are you still interested in working on this, or should we close this ("put it into the attic")?

Feel free to close for now. Probably smarter to split this up a bit.

**Pull request closed without merging**